stage1_fly/run/main.go

```diff
@@ -0,0 +1,381 @@
+		if err != nil {
+			return "", fmt.Errorf("unable to find absolute path for %s", binPath)
+		}
+		d, err := os.Stat(binAbsPath)
+		if err != nil {
+			continue
+		}
+		// Check the executable bit, inspired by os.exec.LookPath()
+		if m := d.Mode(); !m.IsDir() && m&0111 != 0 {
+			return binAbsPath, nil
+		}
+	}
+	return "", fmt.Errorf("unable to find %q in %q", bin, paths)
+}
+
+func withClearedCloExec(lfd int, f func() error) error {
+	err := sys.CloseOnExec(lfd, false)
+	if err != nil {
+		return err
+	}
+	defer sys.CloseOnExec(lfd, true)
+
+	return f()
+}
+
+func writePpid(pid int) error {
+	// write ppid file as specified in
+	// Documentation/devel/stage1-implementors-guide.md
+	out, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("Cannot get current working directory: %v\n", err)
+	}
+	// we are the parent of the process that is PID 1 in the container so we write our PID to "ppid"
+	err = ioutil.WriteFile(filepath.Join(out, "ppid"),
+		[]byte(fmt.Sprintf("%d\n", pid)), 0644)
+	if err != nil {
+		return fmt.Errorf("Cannot write ppid file: %v\n", err)
+	}
+	return nil
+}
+
+func evaluateMounts(rfs string, app string, p *stage1commontypes.Pod) ([]flyMount, error) {
+	imApp := p.Images[app].App
+	namedVolumeMounts := map[types.ACName]volumeMountTuple{}
+
+	for _, m := range p.Manifest.Apps[0].Mounts {
+		_, exists := namedVolumeMounts[m.Volume]
+		if exists {
+			log.Fatalf("fly: duplicated mount given: %q", m.Volume)
+		}
+		namedVolumeMounts[m.Volume] = volumeMountTuple{M: m}
+		log.Printf("Adding %+v", namedVolumeMounts[m.Volume])
+	}
+
+	// Merge command-line Mounts with ImageManifest's MountPoints
+	for _, mp := range imApp.MountPoints {
+		tuple, exists := namedVolumeMounts[mp.Name]
+		switch {
+		case exists && tuple.M.Path != mp.Path:
+			return nil, fmt.Errorf("fly: conflicting path information from mount and mountpoint %q", mp.Name)
+		case !exists:
+			namedVolumeMounts[mp.Name] = volumeMountTuple{M: schema.Mount{Volume: mp.Name, Path: mp.Path}}
+			log.Printf("Adding %+v", namedVolumeMounts[mp.Name])
+		}
+	}
+
+	// Insert the command-line Volumes
+	for _, v := range p.Manifest.Volumes {
+		// Check if we have a mount for this volume
+		tuple, exists := namedVolumeMounts[v.Name]
+		if !exists {
+			return nil, fmt.Errorf("fly: missing mount for volume %q", v.Name)
+		} else if tuple.M.Volume != v.Name {
+			// assertion regarding the implementation, should never happen
+			return nil, fmt.Errorf("fly: mismatched volume:mount pair: %q != %q", v.Name, tuple.M.Volume)
+		}
+		namedVolumeMounts[v.Name] = volumeMountTuple{V: v, M: tuple.M}
+		log.Printf("Adding %+v", namedVolumeMounts[v.Name])
+	}
+
+	// Merge command-line Volumes with ImageManifest's MountPoints
+	for _, mp := range imApp.MountPoints {
+		// Check if we have a volume for this mountpoint
+		tuple, exists := namedVolumeMounts[mp.Name]
+		if !exists || tuple.V.Name == "" {
+			log.Fatalf("fly: missing volume for mountpoint %q", mp.Name)
```

Comment:
`log` are not printed without `--debug` so I get silent errors.

Other errors in the normal stage1 use this pattern:
```
fmt.Fprintf(os.Stderr, ...
return 1

.... os.Exit(stage1())
```